components/script/dom/document.rs

```diff
@@ -518,6 +519,8 @@ pub(crate) struct Document {
     inherited_insecure_requests_policy: Cell<Option<InsecureRequestsPolicy>>,
     /// <https://w3c.github.io/IntersectionObserver/#document-intersectionobservertaskqueued>
     intersection_observer_task_queued: Cell<bool>,
+    /// Active intersection observers that should be processed by this document.
+    intersection_observers: DomRefCell<Vec<Dom<IntersectionObserver>>>,
```

Comment:
I think it would be good to link to https://w3c.github.io/IntersectionObserver/#run-the-update-intersection-observations-steps

and add the prose from step 1, 

and also a note that this is discussed at  https://github.com/w3c/IntersectionObserver/issues/525